AIDevGallery/Samples/Open Source Models/Multimodal Models/DescribeImage.xaml.cs

```diff
@@ -153,68 +159,100 @@ private async IAsyncEnumerable<string> InferStreaming(string question, string im
         }
     }
 
-    private async void Button_Click(object sender, RoutedEventArgs e)
+    private async void LoadButton_Click(object sender, RoutedEventArgs e)
     {
-        if (_cts != null)
+        var file = await PickFileAsync();
+        if (file != null)
         {
-            _cts.Cancel();
-            SelectImageButton.IsEnabled = false;
-            ButtonTextBlock.Text = "Canceling";
-            return;
+            imageFile = file;
+            LoadImage(imageFile);
+
+            await DescribeTheImage(imageFile);
         }
+    }
 
-        var file = await PickFileAsync();
-        if (file != null)
+    private async void Button_Click(object sender, RoutedEventArgs e)
+    {
+        if (ButtonTextBlock.Text == "Run sample" && imageFile != null)
         {
-            SelectImageButton.Focus(FocusState.Programmatic);
-            await LoadAndDescribeImage(file);
+            await DescribeTheImage(imageFile);
+        }
+        else if (ButtonTextBlock.Text == "Cancel" && _cts != null)
+        {
+            CancelGeneration();
         }
     }
 
-    private async Task LoadAndDescribeImage(StorageFile file)
+    private void CancelGeneration()
     {
-        ButtonTextBlock.Text = "Cancel";
-        ToolTipService.SetToolTip(SelectImageButton, "Cancel generation");
-        AutomationProperties.SetName(SelectImageButton, "Cancel generation");
-        Output.Text = string.Empty;
-        Loader.IsActive = true;
-        Loader.Visibility = Visibility.Visible;
+        _cts!.Cancel();
+        DescribeImageButton.IsEnabled = false;
+        ButtonTextBlock.Text = "Canceling, this could take a while";
+    }
 
-        _cts = new CancellationTokenSource();
+    private async void LoadImage(StorageFile file)
+    {
+        using IRandomAccessStream fileStream = await file.OpenAsync(FileAccessMode.Read);
+        BitmapImage bitmapImage = new();
+        await bitmapImage.SetSourceAsync(fileStream);
+        DefaultImage.Source = bitmapImage;
+        NarratorHelper.AnnounceImageChanged(DefaultImage, "Image changed: new upload"); // <exclude-line>
+        imageFile = file;
+    }
 
-        using (IRandomAccessStream fileStream = await file.OpenAsync(FileAccessMode.Read))
+    private async Task DescribeTheImage(StorageFile file)
```

Comment:
I would remove this argument. It's always called only with the internal `imageFile` field.